proposed/2023/Add-dotnet-search.md

```diff
@@ -41,23 +44,87 @@ This option will specify a list of sources to search from. If a source is not sp
 
-Source: nuget.org
-| Package ID                                  | Latest Version | Authors | Downloads       |
-|---------------------------------------------|----------------|---------|-----------------|
-| Newtonsoft.Json                             | 13.0.3         |         | 3,829,822,911   |
+        Source: nuget.org
+        | Package ID                                  | Latest Version | Authors | Downloads       |
+        |---------------------------------------------|----------------|---------|-----------------|
+        | Newtonsoft.Json                             | 13.0.3         |         | 3,829,822,911   |
+
+#### **Option `--format`**
+
+This option will allow the specification of the output format. The option will take one of these three arguments: `table`, or `json`. The outputs will depend on verbosity. The following examples are for normal verbosity.
+
+- `--format table` the output will be in a tabular form as shown below
+
+        Source: nuget.org
+        | Package ID                                  | Latest Version | Authors | Downloads       |
+        |---------------------------------------------|----------------|---------|-----------------|
+        | Newtonsoft.Json                             | 13.0.3         |         | 3,829,822,911   |
+
+- `--format json` : A json output will be outputted. It will contain the following metadata.
+  - Source
+  - Package ID
+  - LatestVersion
+  - Authors
+  - Downloads
+
+  The output will take the following format
+
+        [
+            {
+                "source": "source Name1",
+                "packagesList": [
+                    {
+                        "author": "",
+                        "downloads": "",
+                        "latestVersion": "",
+                        "packageId": ""
+                    }
+                ]
+            },
+            {
+                "source": "source Name2",...
+            }
+        ]
+
+#### Option `--verbosity`
+
+  controls the amount of information included in the output of a command. This allows you to tailor the output to your needs, providing either a concise summary or a detailed breakdown of the information returned by the command.
+
+Here are the details of each verbosity level:
+
+- **`--verbosity minimal`** : The output will contain minimal information. For each package, it will display only the following metadata:
+  - Package ID
+  - Latest Version
+  
+- **`--verbosity normal`** : The output will provide a moderate amount of information. For each package, it will display the following metadata:
+  - Package ID
+  - Latest Version
+  - Authors
+  - Downloads
+  
+- **`--verbosity detailed`** : The output will be highly detailed. For each package, it will display the following metadata:
+  - Package ID
+  - Latest Version
+  - Authors
+  - Downloads
+  - Description
+  - Deprecation
+  - Project URL
+  - Vulnerable
```

Comment:
How much vulnerability info will be shown? Seems like it could be inspired by dotnet list package and how it renders vulnerability info.

Reply:
I was thinking a binary information(True/False)

Reply:
On a second thought, listing the vulnerabilities might be a better idea. As claiming a package is not vulnerable could be misleading.